modules/apps/questions/questions-web/src/main/java/com/liferay/questions/web/internal/portlet/route/QuestionsFriendlyURLMapper.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.questions.web.internal.portlet.route;
+
+import com.liferay.portal.kernel.portlet.DefaultFriendlyURLMapper;
+import com.liferay.portal.kernel.portlet.FriendlyURLMapper;
+import com.liferay.questions.web.internal.constants.QuestionsPortletKeys;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Javier Gamarra
+ */
+@Component(
+	property = {
+		"com.liferay.portlet.friendly-url-routes=META-INF/friendly-url-routes/routes.xml",
+		"javax.portlet.name=" + QuestionsPortletKeys.QUESTIONS
+	},
+	service = FriendlyURLMapper.class
+)
+public class QuestionsFriendlyURLMapper extends DefaultFriendlyURLMapper {
```

Comment:
I guess this way you override the portal url mapping and let React handle all the routing?

Reply:
Not exactly. An [URL mapper](https://help.liferay.com/hc/en-us/articles/360018150851-Adding-Friendly-URL-Mapping-to-the-Portlet) let you configure paths like `http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_journal_web_portlet_JournalPortlet&p_p_lifecycle=0&p_p_state=maximized&p_v_l_s_g_id=33726&p_p_auth=ZWt4266N` that have a lot of parameters and liferay stuff and clean it to something like `localhost:8080/guest/-/MAPPING/` and implicit parameters. 

But here we are changing the mode to contains so it acts differently, it matches the `questions/sectionTitle` URLs and returns nothing.

Reply:
🤔  Understood the first part. The second not so much...